core/spot_instance_request.go

```diff
@@ -19,6 +19,39 @@ type spotInstanceRequest struct {
 	asg    *autoScalingGroup
 }
 
+func (s *spotInstanceRequest) isRequestOpen() bool {
+	return s.State != nil && *s.State == "open"
+}
+
+// Before waiting for an instance we check if amazon has put the open request
+// in a holding state (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-bid-status.html).
+func (s *spotInstanceRequest) isHoldingRequest() bool {
+	return s.isRequestOpen() && s.Status != nil && s.Status.Code != nil && hasHoldingRequestStatus(*s.Status.Code)
+}
+
+func hasHoldingRequestStatus(code string) bool {
```

Comment:
A small nitpick:

We can merge these case statements like this: https://github.com/golang/go/wiki/Switch#multiple-cases

Reply:
I'll have a look to see if I can combine the branches better.  Not sure if multiple-cases is possible, as it's not switching on the single value and check it.  It's different "switch" values.  I can probably pass `s.Status` to `hasHoldingRequestStatus(..)` and let it do it's thing.

Will look when I update the to master tonight. 
cheers
/dom